plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -468,8 +468,8 @@ def static_env
         env[:BLUE_GREEN] = blue_green_color if blue_green_color
 
         # env from plugins
-        plugin_envs = Samson::Hooks.fire(:deploy_group_env, project, @doc.deploy_group, resolve_secrets: false)
-        plugin_envs += Samson::Hooks.fire(:deploy_env, @doc.kubernetes_release.deploy) if @doc.kubernetes_release.deploy
+        deploy = @doc.kubernetes_release.deploy || Deploy.new(project: project)
+        plugin_envs = Samson::Hooks.fire(:deploy_env, deploy, @doc.deploy_group, resolve_secrets: false)
```

Comment:
do you still need to pass through a separate deploy_group? can't you get that from the deploy?

Reply:
deploys don't have a deploy-group, they only know their stage which has multiple

Reply:
would the deploy's deploy group ever not match the stages?

Reply:
oh, this is for a single deploy group? that makes more sense...